rcl/src/rcl/arguments.c

```diff
@@ -267,6 +301,26 @@ rcl_parse_arguments(
         continue;
       }
 
+      // Attempt to parse argument as parameter override flag
+      if (strcmp(RCL_PARAM_FLAG, argv[i]) == 0 || strcmp(RCL_SHORT_PARAM_FLAG, argv[i]) == 0) {
+        if (i + 1 < argc) {
+          // Attempt to parse next argument as parameter override rule
+          if (RCL_RET_OK == _rcl_parse_param_rule(argv[i + 1], args_impl->parameter_overrides)) {
+            RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "param override rule : %s\n", argv[i + 1]);
+            i += 1;  // Skip flag here, for loop will skip rule.
+            continue;
+          } else {
+            RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME,
+              "Couldn't parse arg %d (%s) as parameter override rule. Error: %s", i + 1,
+              argv[i + 1], rcl_get_error_string().str);
+          }
+        } else {
+          RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME,
+            "Couldn't parse arg %d (%s) as parameter override flag. No rule found.\n",
+            i, argv[i]);
+        }
+      }
```

Comment:
Command line parsing in `rcl` does no error handling on its own, beyond filling up the unparsed arguments bucket. I consciously did the same here, to keep this PR as simple as it can be, but I wonder if we should add it (and then figure out how to propagate those errors for downstream code to deal with, maybe re-using `rcutils` error messages?).